tabletop/core/game.js

```diff
@@ -293,4 +307,186 @@ Game.prototype.destroyToken = function(token) {
   this.board.destroyToken(token);
 };
 
-module.exports = Game;
\ No newline at end of file
+
+Game.prototype.pickAImove = function(player, game) { 
+
+  var moves = this.getValidMoves(player);
+  var results = [];
+
+  // for each token, grab the appropriate objects from the game copy object
+  // (since getValidMoves will return actual game objects and we want 
+  // the copies). Then execute the move and score it.
+  for (var i = 0; i < moves.tokens.length; i++) {
+
+    var tokenIdx = this.board.tokens.indexOf(moves.tokens[i]);
+    var tilePos = this.board.getTilePosition(moves.tiles[i]);
+    
+    moves.destinationTiles[i].forEach(function(dest) { 
+      var destPos = this.board.getTilePosition(dest);
+
+      var gameCopy = this.copyGameStatus(this);
+
+      var tokenCp = gameCopy.board.tokens[tokenIdx];
+      var tileCp, destCp;
+      if (game.board instanceof Gridboard) { 
+        tileCp = gameCopy.board.tiles[tilePos.x][tilePos.y];
+        destCp = gameCopy.board.tiles[destPos.x][destPos.y];
+      } else { 
+        tileCp = gameCopy.board.tiles[tilePos];
+        destCp = gameCopy.board.tiles[destPos];
+      } 
+        
+      gameCopy.proposedMove = {
+        token: tokenCp,
+        tile: tileCp,
+        destination: destCp,
+        tokenIdx: tokenIdx, 
+        tilePos: tilePos, 
+        destPos: destPos
+      };
+      
+      gameCopy.executeMove();
+      results.push({
+        score: gameCopy.scoreBoard(player), 
+        token: tokenCp, 
+        tile: tileCp, 
+        destination: destCp,
+        tokenIdx: tokenIdx, 
+        tilePos: tilePos, 
+        destPos: destPos
+      });
+    }, this);
+  }
+  
+  var bestResult = results[0];
+  var bestScore = 0;
+  
+  results.forEach(function(result) { 
```

Comment:
@andrewjmeier can lodash help here? i don't need the max object, i need the max object.score